cpp/src/slice2cpp/Gen.cpp

```diff
@@ -1070,75 +1067,73 @@ Slice::Gen::MetadataVisitor::visitOperation(const OperationPtr& p)
 void
 Slice::Gen::MetadataVisitor::visitDataMember(const DataMemberPtr& p)
 {
-    StringList metadata = validate(p->type(), p->getMetadata(), p->file(), p->line());
-    p->setMetadata(metadata);
+    p->setMetadata(validate(p->type(), p->getMetadata(), p->file(), p->line()));
 }
 
 void
 Slice::Gen::MetadataVisitor::visitSequence(const SequencePtr& p)
 {
-    StringList metadata = validate(p, p->getMetadata(), p->file(), p->line());
-    p->setMetadata(metadata);
+    p->setMetadata(validate(p, p->getMetadata(), p->file(), p->line()));
 }
 
 void
 Slice::Gen::MetadataVisitor::visitDictionary(const DictionaryPtr& p)
 {
-    StringList metadata = validate(p, p->getMetadata(), p->file(), p->line());
-    p->setMetadata(metadata);
+    p->setMetadata(validate(p, p->getMetadata(), p->file(), p->line()));
 }
 
 void
 Slice::Gen::MetadataVisitor::visitEnum(const EnumPtr& p)
 {
-    StringList metadata = validate(p, p->getMetadata(), p->file(), p->line());
-    p->setMetadata(metadata);
+    p->setMetadata(validate(p, p->getMetadata(), p->file(), p->line()));
 }
 
 void
 Slice::Gen::MetadataVisitor::visitConst(const ConstPtr& p)
 {
-    StringList metadata = validate(p, p->getMetadata(), p->file(), p->line());
-    p->setMetadata(metadata);
+    p->setMetadata(validate(p, p->getMetadata(), p->file(), p->line()));
 }
 
-StringList
+MetadataList
 Slice::Gen::MetadataVisitor::validate(
     const SyntaxTreeBasePtr& cont,
-    const StringList& metadata,
+    const MetadataList& metadata,
```

Comment:
pass by value and validate in place?

This would avoid a copy when we call this like:

```
p->setMetadata(validate(p, p->getMetadata(), p->file(), p->line()));
```

I guess it would complicate the loop, as we have to use and iterator to remove items from the vector we are iterating... So maybe not worth the added complexity.

Reply:
Implemented as suggested.
It doesn't make the loop much worse at all, except that we have to use iterators. Which is fine.
This is still C++ after all!